### DISCUSSION TITLE
CoApp not recognized in LibreWolf (Firefox derivative)

### DISCUSSION THREAD

**Discussion by PEHamel** [aclap-dev/video-downloadhelper#216] [Help and Q&A] [answered]
### Your question

Running MacOS Big Sur (Version 11.7.10).

LibreWolf recently re-installed to latest version via Homebrew.

Tried to download a video using VDH. Tells me I need the CoApp.

Downloaded and ran the CoApp PKG file:

CoApp not recognized in LibreWolf.

In Terminal, ran the install:

 /Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp install
Installing…
Writing /Users/eric/Library/Application Support/Mozilla/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /Users/eric/Library/Application Support/Chromium/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /Users/eric/Library/Application Support/Google/Chrome Canary/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /Users/eric/Library/Application Support/Google/Chrome/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /Users/eric/Library/Application Support/BraveSoftware/Brave-Browser/NativeMessagingHosts/net.downloadhelper.coapp.json
VdhCoApp : VdhCoApp is ready to be used

CoApp still not recognized in LibreWolf.

Checked Firefox to see if recognized there. It is. Works fine.

From Firefox VDH settings page:

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 3ffe11f2c3ee4a08
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/eric",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-11-gc30f360955"
}

From LibreWolf VDH settings page:

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: nocoapp
key (only 16 first characters): 3ffe11f2c3ee4a08
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: no.

Sounds like I need to copy net.downloadhelper.coapp.json somewhere. Can anybody tell me where?

Thanks.

### Addon & CoApp version and details

From Firefox VDH settings page:

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 3ffe11f2c3ee4a08
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/eric",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-11-gc30f360955"
}

From LibreWolf VDH settings page:

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: nocoapp
key (only 16 first characters): 3ffe11f2c3ee4a08
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: no.

**Comment by FlailoftheLord** [comment, marked as answer]
I was having the same issue on Linux. The issue is with NativeMessaging. Librewolf extensions store their native messaging host configurations under the default directory ~/.mozilla but the browser reads them from its own ~/.librewolf directory, just copy the folder (.mozilla/native-messaging-hosts) over to librewolf's directory and it works.

**Reply by PEHamel** [reply]
Perfect! Thanks. Works like a charm.

To be clear, I copied the NativeMessagingHosts folder from ~/Library/Application Support/Mozilla/ to ~/Library/Application Support/LibreWolf/

**Comment by tfernsle** [comment]
I had the same problem with Waterfox on MacOS Catalina (10.15), and solved it by installing Firefox, re-installing VDHCoAPP, and then copying NativeMessagingHosts folder from the new mozilla folder to my Waterfox profile folder.

**Reply by quimnut** [reply]
cd ~/Library/Application\ Support
ln -s librewolf mozilla

Put FF in the bin :)

**Reply by le-dawg** [reply]
I can confirm that this fix works. more context for the stragglers out there:

1. Install LIbrewolf, Install vdh, install companion app
2. Install Firefox, Install vdh, install companion app
3. Execute symbolic link above
4. Move FireFox to bin

If you used homebrew to install Firefox - DO NOT use it to remove firefox, as the cleanup scripts might remove the symlinks and break the fix.

**Reply by quimnut** [reply]
1. Install librewolf (assuming you do already)
2. add symlink
3. install coapp
4. remove symlink
